rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -87,7 +87,9 @@ module Impl {
     not name.charAt(0).isUppercase() and
     // exclude parameters from functions without a body as these are trait method declarations
     // without implementations
-    not exists(Function f | not f.hasBody() and f.getParamList().getAParam().getPat() = p)
+    not exists(Function f | not f.hasBody() and f.getParamList().getAParam().getPat() = p) and
+    // exclude parameters from function pointers as well, which also lack a body
```

Comment:
I was a bit confused by this comment so I propose this change:
```suggestion
    // exclude parameters from function pointer types (e.g. `x` in `fn(x: i32) -> i32`)
```
* Makes it clear that it's function pointer _types_ (and not in usage of function pointers).
* Includes an example.
* Remove the mention of a body. I think that comment makes more sense for the `Function` case before where the AST node has a `hasBody` predicate that doesn't hold.

Reply:
Thanks, this definitely helps make things clearer.